Dockerfile

```diff
@@ -18,117 +20,129 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
+RUN add-apt-repository ppa:openjdk-r/ppa \
+  && apt-get update \
+  && apt-get install -y openjdk-8-jdk \
+  && rm -rf /var/lib/apt/lists/*
 
 # Setup JAVA_HOME, this is useful for docker commandline
 ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
 RUN export JAVA_HOME
 
-# Install NodeJS
-ADD ./scripts /scripts
-WORKDIR /scripts
-# This is need to install nodejs 4.x otherwise nodejs 0.x will be installed.
-RUN bash setup_node4.sh
-RUN apt-get install -y nodejs
-
-# Install linuxbrew.
-RUN useradd -m -s /bin/bash linuxbrew
-RUN echo 'linuxbrew ALL=(ALL) NOPASSWD:ALL' >>/etc/sudoers
-USER linuxbrew
-WORKDIR /home/linuxbrew
-ENV PATH /home/linuxbrew/.linuxbrew/bin:/home/linuxbrew/.linuxbrew/sbin:$PATH
-ENV SHELL /bin/bash
-RUN yes |ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Linuxbrew/install/master/install)"
-RUN brew doctor || true
-
-# Manually install protobuf with --no-binary flag before installing protoc
-# Python and pip are installed by 'brew install --with-plugins grpc'. We need
-# to install them preemptively, and then install the protobuf package with
-# the --no-binary flag to work around https://github.com/google/protobuf/issues/2895
-RUN brew install python
-RUN pip install --upgrade pip
-RUN pip install protobuf==3.2.0 --no-binary protobuf
-
-# Install protoc and grpc.
-RUN brew tap grpc/grpc
-RUN brew install --with-plugins grpc
-
-# Install Go1.6 from linuxbrew.
-RUN brew install go
-
-USER root
+# Install NodeJS.
+# This installs Node 4 on Ubuntu 16.04.
+RUN apt-get update \
+  && apt-get install -y \
+    nodejs \
+    npm \
+  && rm -rf /var/lib/apt/lists/* \
+  && ln -s /usr/bin/nodejs /usr/local/bin/node
+
+# Install Ruby.
+# This installs Ruby 2.3 on Ubuntu 16.04.
+RUN apt-get update \
+  && apt-get install -y \
+    ruby \
+    ruby-dev \
+  && rm -rf /var/lib/apt/lists/*
+
+# Install Go.
+# This installs Go 1.6 on Ubuntu 16.04.
+RUN apt-get update \
+  && apt-get install -y golang \
+  && rm -rf /var/lib/apt/lists/*
 
+# Download the go protobuf support.
 ENV GOPATH /go
 ENV PATH $GOPATH/bin:/home/linuxbrew/.linuxbrew/opt/go/libexec/bin:$PATH
-RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" && chmod -R 777 "$GOPATH"
-RUN go get -u github.com/golang/protobuf/proto github.com/golang/protobuf/protoc-gen-go
+RUN mkdir -p "$GOPATH/src" "$GOPATH/bin" \
+  && chmod -R 777 "$GOPATH" \
+  && go get -u github.com/golang/protobuf/proto github.com/golang/protobuf/protoc-gen-go
 
 # Install packman
 # TODO: consider installing released packages once artman is versioned (so that
 #   each release of artman is pegged to a release of packman)
 RUN npm install -g https://github.com/googleapis/packman.git
 
 # Setup tools for codegen of Ruby
-RUN gem install rubocop --version '= 0.39.0' --no-ri --no-rdoc
-RUN gem install bundler --version '= 1.12.1' --no-ri --no-rdoc
-RUN gem install rake --version '= 10.5.0' --no-ri --no-rdoc
-RUN gem install grpc-tools --version '=1.0.0' --no-ri --no-rdoc
+RUN gem install rake --no-ri --no-rdoc \
```

Comment:
Installing `rake` was required for the `rubocop` to work with this version of Ruby.